deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/worker/AutoPromoter.java

```diff
@@ -236,6 +236,15 @@ void processOnce(String envId) throws Exception {
             currDeployBean = deployDAO.getById(deployId);
         }
 
+        if (currDeployBean != null) {
+            if (promoteBean.getOverride_policy == PromoteOverridePolicy.NO_OVERRIDE) {
+            // do not override! 
```

Comment:
Can you add some logs here? something like:
LOG.info("override policy is set to no override for env %s, will wait for current deploy to complete")